libraroo-web/src/bookCatalog/components/CreateBookCatalog.tsx

```diff
@@ -0,0 +1,95 @@
+import { useState } from "react";
+import { useNavigate } from "react-router-dom";
+import bookCatalogService  from "../services/bookCatalog.service";
+import { Alert, Box, Button, Container, TextField } from "@mui/material";
+
+export const CreateBookCatalog: React.FC = () => {
+  const [author, setAuthor] = useState("");
+  const [title, setTitle] = useState("");
+  const [language, setLanguage] = useState("");
+  const [error, setError] = useState<string | null>(null);
+  const [loading, setLoading] = useState(false);
+
+  const navigate = useNavigate(); // React Router's navigation hook
+
+  const handleSubmit = async (event: React.FormEvent) => {
+    event.preventDefault();
+    setLoading(true);
+    setError(null); // Clear previous errors
+
+    try {
+      await bookCatalogService .addBookCatalog({
+        author,
+        title,
+        language,
+      });
+
+      // TODO : Navigate to book catalog list screen.
+    } catch (err: any) {
+      setError("An error occured while trying to add book, check and try again.");
```

Comment:
First check if this is an axios error, then print the message from the API.
if not print this mesage